operator/v1/types_network.go

```diff
@@ -561,7 +563,31 @@ type IPsecConfig struct {
 	// this is left to the user to configure.
 	// +kubebuilder:validation:Enum=Disabled;External;Full
 	// +optional
+	// +unionDiscriminator
 	Mode IPsecMode `json:"mode,omitempty"`
+
+	// full defines configuration parameters for the IPsec `Full` mode.
+	// This is permitted only when mode is configured with `Full`,
+	// and forbidden otherwise.
+	// +unionMember,optional
+	Full *IPsecFullModeConfig `json:"full,omitempty"`
```

Comment:
Needs an explicit optional tag too please
```suggestion
	// +optional
	Full *IPsecFullModeConfig `json:"full,omitempty"`
```

Reply:
done.